cloudbuild-test.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+# set the project ID in the k8s manifest for service account/workload identity
+- name: 'ubuntu'
+  id: Replace
+  script: |
+    #!/usr/bin/env bash
+    sed -i "s/%GCLOUD_PROJECT%/${PROJECT_ID}/g" collector/*
+  env:
+  - 'PROJECT_ID=$PROJECT_ID'
+
+# deploy the manifests in GKE
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: Deploy
+  args:
+  - 'apply'
+  - '-f'
+  - './collector/.'
+  env:
+  - 'CLOUDSDK_COMPUTE_REGION=${LOCATION}'
+  - 'CLOUDSDK_CONTAINER_CLUSTER=${_GKE_CLUSTER}'
+
+# assert that all pods are up and running
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: AssertReady
+  args:
+  - 'wait'
+  - '--for=condition=Ready'
+  - '--timeout=300s'
+  - '-n'
+  - 'opentelemetry'
+  - 'pods'
+  - '--all'
+  env:
+  - 'CLOUDSDK_COMPUTE_REGION=${LOCATION}'
+  - 'CLOUDSDK_CONTAINER_CLUSTER=${_GKE_CLUSTER}'
+
+# delete everything in the opentelemetry namespace
```

Comment:
Are you planning to spin up/tear down a fresh GKE cluster for each presubmit run? I think that's fine, just checking because otherwise this will be a problem for concurrent PRs

Reply:
I added steps to bookend the script with creation and deletion of a cluster. I'm using the commit SHA as the name for now, but once we set up the trigger with the presubmits, I might change it to PR number or something better

Reply:
I think SHA is better than PR number, but I actually like using a combination of PR, SHA, and timestamp. That gives the most info about what change triggered a run and allows multiple runs for the same change (like a retest). For example, if a run fails because of a flake and we keep the cluster around to confirm that it was a flake, we can retrigger it.

But, I think for now SHA is perfect 👍